src/packet/many.rs

```diff
@@ -258,7 +258,9 @@ fn read_fixed<R: Read>(
     }
 
     out.resize(out_len + len, 0u8);
-    reader.read_exact(&mut out[out_len..])?;
+    if let Some(ref mut out) = out.get_mut(out_len..) {
```

Comment:
This isn't `if` actually, maybe `let ... else return Err(...);` is better? Some dead code is added anyway, be it `if` or `return Err(...)`, but the latter protects from wrong code modifications at least